### DISCUSSION TITLE
Dynamic new TextInputInterface on click

### DISCUSSION THREAD

**Discussion by Veda1097** [newcat/baklavajs#394] [Q&A]
I have a Port btn on the node
How do I add the functionality such that when user clicks on "Ports" btn,
Dynamically a new TextInputInterface should be added below the node as P1, P2, P3 ...as per no. of clicks

#MyNode.ts / jsx
import { defineNode, NodeInterface } from "@baklavajs/core";
import { NumberInterface, SelectInterface,ButtonInterface } from "@baklavajs/renderer-vue";
 
export default defineNode({
  type: "MyNode",
  inputs: {
    number1: () => new NumberInterface("Number", 1),
    number2: () => new NumberInterface("Number", 10),
    operation: () =>
      new SelectInterface("Operation", "Add", ["Add", "Subtract"]).setPort(
        false
      ),
      port_button: () => new ButtonInterface("Port", () => console.log("Button clicked")),
  },
  outputs: {
    output: () => new NodeInterface("Output", 0)
  }
});

**Comment by newcat** [comment]
This can be done either by using a dynamic node ([example](https://github.com/newcat/baklavajs/blob/master/packages/renderer-vue/playground/DynamicNode.ts)) or a custom node ([example](https://github.com/newcat/baklavajs/blob/master/packages/renderer-vue/playground/AdvancedNode.ts))
